### PR TITLE
feat: add front-end tests

### PR DESCRIPTION
**Pull request by wazelin**
# [FUN-965](https://oat-sa.atlassian.net/browse/FUN-965)

This PR enables client-side test and includes them into the report publication.

See https://github.com/oat-sa/tao-community/actions/runs/5544605113/jobs/10122308704#step:11:444 and https://github.com/oat-sa/tao-community/pull/517#issuecomment-1634380856.
```
Execution Time (2023-07-13 14:55:02 UTC-0)
loading tasks                   1.1s  ▇ 2%
qunit:qtiitempcitest            4.2s  ▇▇ 6%
qunit:taotest                  11.6s  ▇▇▇▇▇▇ 17%
qunit:taoclientdiagnostictest   3.2s  ▇▇ 5%
qunit:taodeliveryrdf           833ms  ▇ 1%
qunit:taoitemstest             873ms  ▇ 1%
qunit:taomediamanagertest       2.1s  ▇ 3%
qunit:taoqtiitemtest           19.5s  ▇▇▇▇▇▇▇▇▇ 28%
qunit:taoqtitest                6.7s  ▇▇▇▇ 10%
qunit:taoqtitestpreviewer      15.7s  ▇▇▇▇▇▇▇▇ 23%
qunit:taotaskqueuetest          1.6s  ▇ 2%
Total 1m 9.1s
```

[FUN-965]: https://oat-sa.atlassian.net/browse/FUN-965?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by azaitsau — APPROVED** [review]

**Review by kilatib — APPROVED** [review]
LGTM

File: .github/workflows/continuous-integration.yaml [hunk 1]
```diff
@@ -43,61 +52,111 @@ jobs:
...
-      - name: Run tests
+      - name: Generate localization files
+        if: ${{ matrix.node-version }}
+        run: php tao/scripts/tools/GenerateTranslationBundles.php
+
+      - name: Front-end tests
+        if: ${{ matrix.node-version }}
+        working-directory: tao/views/build
+        run: |
+          perl -0777 -pi -e "s/(puppeteer[\s\r\n]*:[\s\r\n]*.*args:[\s\r\n]+\[)/\1'--disable-dev-shm-usage','--disable-setuid-sandbox','--single-process',/s" grunt/test.js
```
Comment (oatymart) [inline .github/workflows/continuous-integration.yaml:69]: There must be a saner way to set some args, than editing a file on the fly!

**Review by oatymart — COMMENTED** [review]
Great job on getting FE tests to fully run, and improving the pipeline again!

If you want to stop at this version I don't mind, but if we would spend an extra day or so we could modify the grunt files in tao-core so it can get the good puppeteer args for this environment in a regular way?

File: .github/workflows/continuous-integration.yaml (hunk 1 above)
Comment (wazelin) [reply .github/workflows/continuous-integration.yaml:69]: For sure :)

In fact, I noticed only later that the flags I added probably do not make much of a difference. Let me try to execute the workflow without this ugly line.

Comment (wazelin) [reply .github/workflows/continuous-integration.yaml]: Yep.

The run was successful without these flags as well. So, while they could be adding to stability of the tests, I guess we can live without this ugly hack.

**Review by oatymart — APPROVED** [review]

**Review by azaitsau — APPROVED** [review]

**Pull request merged**